### PR TITLE
feat(docs): add details to the readme

### PR DESCRIPTION
**Pull request by BrennaEpp**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -6,7 +6,76 @@ This repository is used by Storage Client libraries to test integration tests lo
...
+## What is this emulator?
+**Note that the words emulator and testbench are used interchangeably below.
+
+This emulator fakes the Google Cloud Storage (GCS) API. You can configure the GCS client libraries to make calls to this fake rather than to the actual API.
+* The emulator fakes the JSON API (both over REST and gRPC), and has limited support for the XML API.
+* Generally, the error codes are similar to the ones generated by GCS, but the error messages are not. 
+* The emulator performs far fewer error checks and no permission checks (ACL/IAM). 
+
+## When to use this emulator
+In general, this emulator is best suited for integration tests that are hard or even annoying to reliably run against production. The primary example of this are errors that make the client library go through its retry path.
```
Comment (coryan) [inline README.md:33]: s/hard or even annoying/hard (or just annoying)/

File: README.md [hunk 2]
```diff
@@ -6,7 +6,76 @@ This repository is used by Storage Client libraries to test integration tests lo
...
+
+This emulator can be useful to test HMAC keys, which are really hard to test against production due to quota restrictions.
+
+It is useful as well to test features that are not yet deployed to production, you can implement them in the emulator and then write the library code before production is "ready".
```
Comment (coryan) [inline README.md:37]: s/production, you/production: you/

File: README.md [hunk 3]
```diff
@@ -6,7 +6,76 @@ This repository is used by Storage Client libraries to test integration tests lo
...
+## How to use this emulator
+
+### Initial set up
+
+1. [Set up python if you haven't already](https://cloud.google.com/python/docs/setup)
+2. Clone this repository
+3. ```bash
+   cd storage-testbench
+    ```
+5. [Create a virtual environment](https://cloud.google.com/python/docs/setup#installing_and_using_virtualenv)
```
Comment (coryan) [inline README.md:48]: nit: the numbers render correctly, but do not match in the source.

File: README.md [hunk 4]
```diff
@@ -6,7 +6,76 @@ This repository is used by Storage Client libraries to test integration tests lo
...
+This emulator can be useful to test HMAC keys, which are really hard to test against production due to quota restrictions.
+
+It is useful as well to test features that are not yet deployed to production, you can implement them in the emulator and then write the library code before production is "ready".
+
+## How to use this emulator
+
+### Initial set up
+
+1. [Set up python if you haven't already](https://cloud.google.com/python/docs/setup)
+2. Clone this repository
```
Comment (coryan) [inline README.md:44]: how?

File: README.md [hunk 5]
```diff
@@ -6,7 +6,76 @@ This repository is used by Storage Client libraries to test integration tests lo
...
+3. ```bash
```
Comment (coryan) [inline README.md:45]: Sometimes you describe what to do and the command to do it, sometimes just the command, sometimes what do to without command.  Maybe consistently do both?

File: README.md [hunk 6]
```diff
@@ -6,7 +6,76 @@ This repository is used by Storage Client libraries to test integration tests lo
...
+    
+### Run the emulator
+
+Run this command from a terminal:
+
+```bash
+gunicorn --bind "localhost:9000" --worker-class sync --threads 10 --access-logfile - "emulator:run()"
+```
+
+Note: Ensure that the virtual environment you created to install the dependencies is active.
```
Comment (coryan) [inline README.md:63]: Use some formatting to make this note more visible.

File: README.md [hunk 7]
```diff
@@ -6,7 +6,76 @@ This repository is used by Storage Client libraries to test integration tests lo
...
+
+If you want the emulator server to automatically reload on changes to it, add the `--reload` flag:
```
Comment (coryan) [inline README.md:65]: Why not just do that all the time and save yourself some description?

**Review by coryan — CHANGES_REQUESTED** [review]
Thanks for the PR!  Sorry for the plethora of comments, but now that I see them in the README the improvements come to mind.

File: README.md (hunk 4 above)
Comment (cojenco) [reply README.md:44]: If it's okay, I always love directing folks to the [How-to Guide](https://github.com/googleapis/google-cloud-cpp/blob/main/doc/contributor/howto-guide-forks-and-pull-requests.md) in the cpp repo. Or we could just link to [GitHub Docs](https://docs.github.com/en/github/creating-cloning-and-archiving-repositories/cloning-a-repository-from-github/cloning-a-repository).

**Review by coryan — APPROVED** [review]
Thanks!

**Review by BrennaEpp — COMMENTED** [review]
@coryan Thanks for the improvements! Please let me know if you see anything else.

**Pull request merged**